python/ray/rllib/train.py

```diff
@@ -16,36 +17,55 @@
 
 parser = argparse.ArgumentParser(
     description=("Train a reinforcement learning agent."))
-parser.add_argument("--env", required=True, type=str)
-parser.add_argument("--alg", required=True, type=str)
-parser.add_argument("--config", default="{}", type=str)
-parser.add_argument("--upload-dir", default="file:///tmp/ray", type=str)
+parser.add_argument("--redis-address", default=None, type=str,
+                    help="The Redis address of the cluster.")
+parser.add_argument("--env", required=True, type=str,
+                    help="The gym environment to use.")
+parser.add_argument("--alg", required=True, type=str,
+                    help="The reinforcement learning algorithm to use.")
+parser.add_argument("--num-iterations", default=sys.maxsize, type=int,
+                    help="The number of training iterations to run.")
+parser.add_argument("--config", default="{}", type=str,
+                    help="The configuration options of the algorithm.")
+parser.add_argument("--upload-dir", default="file:///tmp/ray", type=str,
+                    help="Where the traces are stored.")
 
 
 if __name__ == "__main__":
     args = parser.parse_args()
+    json_config = json.loads(args.config)
 
-    ray.init()
+    if args.redis_address:
+        # TODO(pcm): This is a hack, provide a better solution
+        if args.alg == "A3C":
+            num_workers = a3c.DEFAULT_CONFIG["num_workers"]
+            num_workers = json_config.get("num_workers", num_workers)
+            ray.init(redis_address=args.redis_address,
+                     num_cpus=num_workers)
+        else:
+            ray.init(redis_address=args.redis_address)
+    else:
+        ray.init()
 
     env_name = args.env
     if args.alg == "PolicyGradient":
         config = pg.DEFAULT_CONFIG.copy()
-        config.update(json.loads(args.config))
+        config.update(json_config)
         alg = pg.PolicyGradient(
             env_name, config, upload_dir=args.upload_dir)
     elif args.alg == "EvolutionStrategies":
         config = es.DEFAULT_CONFIG.copy()
-        config.update(json.loads(args.config))
+        config.update(json_config)
         alg = es.EvolutionStrategies(
             env_name, config, upload_dir=args.upload_dir)
     elif args.alg == "DQN":
         config = dqn.DEFAULT_CONFIG.copy()
-        config.update(json.loads(args.config))
+        config.update(json_config)
         alg = dqn.DQN(
             env_name, config, upload_dir=args.upload_dir)
     elif args.alg == "A3C":
         config = a3c.DEFAULT_CONFIG.copy()
-        config.update(json.loads(args.config))
+        config.update(json_config)
```

Comment:
What do you think about using PPO instead of PolicyGradient?

If we do that, then I'd just make all of the abbreviations, e.g., ES for EvolutionStrategies, but I don't feel strongly about this.